macros/src/codegen/dispatchers.rs

```diff
@@ -73,27 +76,88 @@ pub fn codegen(app: &App, analysis: &Analysis, _extra: &Extra) -> Vec<TokenStrea
                 let fq = util::fq_ident(name);
                 let inputs = util::inputs_ident(name);
                 let (_, tupled, pats, _) = util::regroup_inputs(&task.inputs);
+                let exec_name = util::internal_task_ident(name, "EXEC");
 
-                quote!(
-                    #(#cfgs)*
-                    #t::#name => {
-                        let #tupled =
-                            (&*#inputs
-                            .get())
-                            .get_unchecked(usize::from(index))
-                            .as_ptr()
-                            .read();
-                        (&mut *#fq.get_mut()).split().0.enqueue_unchecked(index);
-                        let priority = &rtic::export::Priority::new(PRIORITY);
-                        #name(
-                            #name::Context::new(priority)
-                            #(,#pats)*
-                        )
-                    }
-                )
+                if task.is_async {
+                    let executor_run_ident = util::executor_run_ident(name);
+
+                    quote!(
+                        #(#cfgs)*
+                        #t::#name => {
+                            if !(&mut *#exec_name.get_mut()).is_running() {
+                                let #tupled =
+                                    (&*#inputs
+                                    .get())
+                                    .get_unchecked(usize::from(index))
+                                    .as_ptr()
+                                    .read();
+                                (&mut *#fq.get_mut()).split().0.enqueue_unchecked(index);
+
+                                let priority = &rtic::export::Priority::new(PRIORITY);
+                                (&mut *#exec_name.get_mut()).spawn(#name(#name::Context::new(priority), #(,#pats)*));
```

Comment:
there's one extra comma:

```suggestion
                                (&mut *#exec_name.get_mut()).spawn(#name(#name::Context::new(priority) #(,#pats)*));
```

Reply:
Nice Catch! That would make task arguments stop working.